core/src/main/java/com/linecorp/armeria/common/logging/ClientConnectionTimings.java

```diff
@@ -181,6 +187,37 @@ public long pendingAcquisitionDurationNanos() {
         return pendingAcquisitionDurationNanos;
     }
 
+    /**
+     * Returns the time when the client started checking if an existing connection can be acquired,
+     * in microseconds since the epoch.
+     *
+     * @return the start time, or {@code -1} if there was no action to get a pending connection.
+     */
+    public long existingAcquisitionStartTimeMicros() {
```

Comment:
What do you think of implementing `acquisition_type` mentioned in https://github.com/line/armeria/issues/4886#issuecomment-1558328036 as an API?
Otherwise, this kind of API makes users visit all related methods to get the actual connection acquired duration.
```java
var acutalAcquisitionDuration;
if (timing.connectionAcquisitionDurationNanos() > -1) {
  acutalAcquisitionDuration = timing.connectionAcquisitionDurationNanos();
} else if (timing.pendingAcquisitionDurationNanos() > -1) {
  acutalAcquisitionDuration = timing.pendingAcquisitionDurationNanos();
} else if (timing.existingAcquisitionDurationNanos() -1) {
  acutalAcquisitionDuration = timing.existingAcquisitionDurationNanos();
}
```

I prefer to deprecate `pendingAcquisitionXXX` methods, not add `existingAcquisitionXXX` and expose `ConnectionAcquisitionType` as an enum.

Reply:
> I prefer to deprecate pendingAcquisitionXXX methods, not add existingAcquisitionXXX and expose ConnectionAcquisitionType as an enum.

I thought this would be too big of a breaking change to introduce in a minor release 😅 
To be clear, are you proposing that we deprecate the following methods/metrics
- dnsResolution
- socketConnect
- pendingAcquisition
- existingAcquisition
and provide them as an enum?

I think it's doable, but I wasn't sure what we would gain compared to the breaking change.

> Otherwise, this kind of API makes users visit all related methods to get the actual connection acquired duration.

Couldn't users just look at `connectionAcquisitionDuration` for this information?


I do think it may be a cleaner API to represent the timing hierarchy information via API, but I'm not sure how realistic this is. For instance, if we have an additional hierarchy (i.e. `pendingAcquisition` is again dividable) I'm not sure if it's worth making another breaking change.

Reply:
> Couldn't users just look at connectionAcquisitionDuration for this information?

I was confused. I thought `connectionAcquisitionDuration` becomes -1 when `pendingAcquisition` is set.

`socketConnect`, `pendingAcquisition` and `existingAcquisition` have the same level. One of them should be measured after DNS resolution if no IP address is given. If one value is set, other values must be -1.

`dnsResolution` is not mutually exclusive to the timing information above. So I don't think we need to provide `dnsResolution` as an enum.

Reply:
After checking the piggyback channel acquisition logic, I realized that `socketConnect` can be recorded if the previous acquisition failed. As the values are not mutually exclusive, I have no reason to insist on using `ClientConnectionTimingsType`.

Sorry for the wrong direction. Please ignore the original comment.

Reply:
No worries 😄   Let me re-request reviews after reverting changes